recommendations.md

```diff
@@ -0,0 +1,236 @@
+
+> Customers paying for "Premier Support" get additional access to our support team with a guaranteed SLA.
+
+- Good: https://ubuntu.com/security/esm (Clearly explains what ESM means for Ubuntu end users)
+
+## Versioning Policy
+
+Document your versioning policy. Even if the policy is homebrew and varies between major versions, a clearly
+documented policy is better than none.
+
+Good:
+
+> We follow Semantic Versioning, and limit breaking changes to major upgrades.
+
+> The project follows SemVer starting from v12. Prior releases may include breaking changes in minor version upgrades.
+
+## Release Notes
+
+Release notes are critical for your users doing actual upgrades. If certain upgrade pathways are unsupported
+(such as doing 2 major upgrades at once), document it. Highlight breaking changes in your release notes.
+
+If you have a migration guide, link it on all the release notes.
+
+## Listing Releases
+
+List your releases in a table with all the relevant information for each release cycle. This includes:
+
+1. Link to a changelog.
+2. What's the latest release in that cycle.
+3. What are the supported dates for this release (for *all* different support levels)
+4. Download URL, if needed
+5. Release notes
+6. Migration Guide, if available
+
+Prefer listing older/unsupported releases elsewhere (`/historical-releases`). If you think they are important to your
+users, mark them extremely well in the table as unsupported.
+
+## Dates
+
+### Always use absolute dates
+
+Many times, your support/EoL policies are relative. Common examples:
+
+1. The last major release becomes unsupported 90 days after a new major release.
+2. Bug fixes on previous releases will be made till the latest releases gets the first point release.
+
+However, none of this is relevant to your end-users. Make sure that all your releases always have a clear dates
+(I suggest `YYYY-MM-DD`) irrespective of how these dates are decided. You doing the math once will save your users much
+more time.
+
+### Bad:
+
+| K8s version | AKS GA   | End of life |
+|-------------|----------|-------------|
+| 1.22        | Nov 2021 | 1.25 GA     |
+| 1.23        | Feb 2022 | 1.26 GA     |
+
+(Source: [Azure Kubernetes Release Calendar][aks])
+
+### Bad:
+
+|Version|Release Date|
+---|---
+2.1|3rd March 2021
+2.0|1st March 2020
+
+Release are supported for 2 years from the release date.
+
+### Good
+
+Same as above, but we do the math:
+
+Version|Release Date|EoL Date
+---|---|---
+2.1|3rd March 2021|3rd March 2023
+2.0|1st March 2020|1st March 2022
+
+
+### Good:
+
+|Kubernetes version|Upstream release|Amazon EKS release|Amazon EKS end of support|
+|------|-------------------|-------------------|--------------------|
+| 1.20 | December 8, 2020  | May 18, 2021      | July, 2022         |
+| 1.21 | April 8, 2021     | July 19, 2021     | September, 2022    |
+
+(Source: [Amazon EKS Release Calendar][eks])
+
+## Provide Complete Date
```

Comment:
```suggestion
## Provide complete dates
```

Reply:
Do you have a better suggestion for "complete date"?